app/assets/javascripts/controllers/ems_common/ems_common_form_controller.js

```diff
@@ -309,8 +309,8 @@ ManageIQ.angular.app.controller('emsCommonFormController', ['$http', '$scope', '
       ($scope.emsCommonModel.emstype == "ec2" ||
        ['kubevirt', 'nuage_network', 'openstack', 'openstack_infra', 'rhevm', 'scvmm', 'vmwarews', 'vmware_cloud'].includes($scope.emsCommonModel.emstype) &&
        $scope.emsCommonModel.default_hostname) &&
-      ($scope.emsCommonModel.default_userid != '' && $scope.angularForm.default_userid.$valid &&
-       $scope.emsCommonModel.default_password != '' && $scope.angularForm.default_password.$valid)) {
+      ($scope.emsCommonModel.default_userid != '' && $scope.angularForm.default_userid !== undefined && $scope.angularForm.default_userid.$valid &&
```

Comment:
why do we need to check for `ssh_keypair_userid === undefined` for if the tab is correctly displayed?

Reply:
(Either looks like a bug that needs to be fixed in the other places, or a random change that could potentially break other places.)

Reply:
This was the fix that allowed me to remove the earlier change of "ng-if" to "ng-show". Without the 'undefined' check, this code fails with a javascript error. We didn't see the error appear until adding the sshkey tab, since the other case where the sshkey tab is used (ems_infra) uses 'ng-show' instead of 'ng-if' to display the tab. I followed the example of the existing code (see below in this same method) for handling fields which may be undefined. If the prior change using 'ng-show' instead of 'ng-if' is better, I could revert back to the prior version, but I was following your suggestion here to fix this method rather than using ng-show in the view. I think this version is probably a cleaner solution than what was done before. I suspect getting rid of the problem without checking for undefined would be a much larger change to this code which would be tricky to make sure it didn't break other things (across providers, etc.). This is a pretty safe change, since it basically can't break anything -- for any case where these are undefined, the code is broken without this change already, and when these are defined, checking for undefined is safe.

Reply:
OK, makes sense, thanks :) I was just trying to make sure this is still intended :)